### ISSUE TITLE
After setting a variable to undefined, refreshing the page sets it to null

### ISSUE THREAD

**Issue by AcademyOfF** [tmedwards/sugarcube-2#90]
```
:: StoryData
{
        "ifid": "5CA2DDDE-6270-44F3-BEE4-ADEE564275CB"
}

:: StoryTitle
Sandbox

:: Start
<<set $my_var to 5>>
$$my_var = $my_var

<<button "Continue" "Test Test2">><</button>>

:: Test Test2
Click the button, then refresh the page. You will see that it is set to null, not undefined.
<<run alert($my_var)>>

<<button "Set to undefined" `passage()`>>
  <<set $my_var to undefined>>
<</button>>
```
using the unset macro does not reproduce this issue.

Expected behavior: refreshing the page should alert `undefined`.

**Comment by ChapelR** [comment]
This is likely because `undefined` can't be serialized as it's not a valid JSON type, and is therefore not *really* supported by SugarCube. Still, it would probably be better if the value were deleted instead of being massaged into a `null`. 

As a sort of stylistic rule though, `null` is what is generally recommended for when you want to set a value to "nothing" as `undefined` is used when the value has never been set in the first place. If you want to make something `undefined`, usually the `delete` operator (or `<<unset>>`) is used.

**Comment by AcademyOfF** [comment]
> Still, it would probably be better if the value were deleted instead of being massaged into a null

Yep, that's the behavior I would want.

**Comment by greyelf** [comment]
> Yep, that's the behavior I would want.

If that is the case then why do you do a `<<set $my_var to undefined>>` instead of a `<<unset $my_var>>` ?

**Comment by AcademyOfF** [comment]
I'm saying this as fallback behavior.  I'd prefer it to remain set to undefined, but I'm accepting that JSON doesn't allow that.

**Comment by tmedwards** [comment]
Resolved by b64afa6.
